### ISSUE TITLE
make elasticsearch output for logstash modules configurable 

### ISSUE THREAD

**Issue by jakommo** [elastic/logstash#8853]
Currently the netflow module creates daily indices and this is hard coded [here](https://github.com/elastic/logstash/blob/master/modules/netflow/configuration/logstash/netflow.conf.erb#L608) and [here](https://github.com/elastic/logstash/blob/b4ad9285320a3431441f0364530854432c14e583/logstash-core/lib/logstash/modules/logstash_config.rb#L65-L83).
Since oversharding is a huge problem, especially with logging use cases, it would be nice to make this configurable, so one could use monthly indices or the rollover API.
I know that I can set the template to only create one shard, but if data volume is low, one shard per day can be still too much.

**Comment by jordansissel** [comment]
Let's zoom out a bit to a few areas for discussion, perhaps in another ticket (since this one is focused specifically on "netflow elasticsearch indices"):

1) How configurable do we intend modules to be?

And for configuration, can we make things configurable but also not complex? Arbitrary indexing strategy is going to be very complex, and I'd rather just give users the Logstash configuration to modify instead of inventing a different syntax just for modules.

2) Who is the target audience of a module?

Our target audience, at least as designed, were first-time-getting-started users, so it makes sense that we don't have much configuration options. 

3) Should all modules get "indexing strategy" configuration settings?

If so, what would this look like? I would prefer *not* deploying "configurable index strategyy for netflow" if we don't also have it for arcsight and any other modules.

---

Specific to rollover, which we cannot use today due to limitations in Elasticsearch, @jsvd what do you think about adding some of Curator's functionality into the Elasticsearch output? We already do template management, adding scheduled rollover seems fine to me, and this would help us move to use rollover indices by default.

**Comment by jakommo** [comment]
> since this one is focused specifically on "netflow elasticsearch indices"):

I hit it with the netflow module, but I think it would make sense to have this in all modules.

> How configurable do we intend modules to be?
> And for configuration, can we make things configurable but also not complex? Arbitrary indexing strategy is going to be very complex, and I'd rather just give users the Logstash configuration to modify instead of inventing a different syntax just for modules.

I think it would be enough to just expose the [index setting](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-index) of the Elasticsearch output. 
It could still create daily indices by default (as the ES output does), but it would give users the option to adjust to their indexing scheme. 

> Who is the target audience of a module?
> Our target audience, at least as designed, were first-time-getting-started users, so it makes sense that we don't have much configuration options.

Sure, if we stick with the default of daily indices, I feel like it won't change anything for new users, while giving more advanced users the option to adjust it to their needs. 

> Should all modules get "indexing strategy" configuration settings?
> If so, what would this look like? I would prefer not deploying "configurable index strategyy for netflow" if we don't also have it for arcsight and any other modules.

If we expose the [index setting](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-index) I would do so on all modules.

I think there might be a misunderstanding, I'm not asking for a whole lot "indexing strategy" settings here, I just would like to see the index setting of the Elasticsearch output to be exposed, so one could use weekly or monthly indices instead of daily. 

Regarding rollover, I didn't mean to ask for this functionality being added to the ES output (though it sounds like a cool thing), I basically just used this as a second example of making `index =>` configurable. I.e. when using rollover, you need to point indexing to an alias (e.g. netflow) and this alias points to the latest index, which is not possible when `index =>` can not be configured. 

To sum up, my ask here is to expose the [index setting](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-index) of the Elasticsearch output to be configurable when using modules.

**Comment by PeterWeiler** [comment]
+1 for the ability to provide configuration to the elasticsearch output of the logstash module

In my use case, the volume of flow data is sufficient to require hourly indices and/or additional shards.  At present I need to manually edit [this file](https://github.com/elastic/logstash/blob/b4ad9285320a3431441f0364530854432c14e583/logstash-core/lib/logstash/modules/logstash_config.rb#L65-L83), which is certainly undesirable as it may break/revert on upgrade of logstash.

**Comment by PornCoinVal** [comment]
Can you add variable "netflow.var.index.name=name".
Exampel:
network device 1 (Cisco ASA) - var.udp.port=9995 && var.index.name=netflow-asa for user 1
network device 2 (Mikrotik) - var.udp.port=9996 && var.index.name=netflow-mk for user 2
network device N(...)- var.udp.port=NNNN && var.index.name=netflow-mk for user NNN
